suite-native/receive/src/components/UnverifiedAddressDevice.tsx

```diff
@@ -1,103 +1,113 @@
-// more: https://github.com/trezor/trezor-suite/issues/9778
-const DEVICE_SCREEN_BACKGROUND_COLOR = '#000000';
-const DEVICE_TOP_PADDING = 30;
 const DEVICE_SCREEN_WIDTH = 340;
-const DEVICE_SCREEN_HEIGHT = 150;
-const DEVICE_TEXT_COLOR = '#FFFFFF';
-const FONT_SIZE = 18;
-const LINE_HEIGHT = 32;
-const LINE_WIDTH = 210;
-const FADEOUT_GRADIENT_CENTER = LINE_WIDTH / 2;
-const FADEOUT_GRADIENT_HEIGHT = LINE_HEIGHT * 2.75;
+const DEVICE_SCREEN_HEIGHT = 200;
+
+const PAN_GESTURE_THRESHOLD = 50;
 
 const deviceFrameStyle = prepareNativeStyle(utils => ({
+    width: DEVICE_SCREEN_WIDTH,
     padding: utils.spacings.extraSmall,
     borderWidth: utils.borders.widths.small,
     borderRadius: utils.borders.radii.large / 2,
     borderColor: utils.colors.borderOnElevation1,
 }));
 
-const deviceScreenStyle = prepareNativeStyle(utils => ({
-    justifyContent: 'center',
-    alignItems: 'center',
-    maxWidth: DEVICE_SCREEN_WIDTH,
-    backgroundColor: DEVICE_SCREEN_BACKGROUND_COLOR,
-    borderRadius: utils.borders.radii.large / 2,
-
-    paddingVertical: DEVICE_TOP_PADDING,
-}));
-
-const deviceContentCanvasStyle = prepareNativeStyle(_ => ({
-    width: LINE_WIDTH,
-    height: DEVICE_SCREEN_HEIGHT,
-}));
-
-const AddressLineGradient = ({ isApplied }: { isApplied: boolean }) => (
-    <LinearGradient
-        start={vec(FADEOUT_GRADIENT_CENTER, 0)}
-        end={vec(FADEOUT_GRADIENT_CENTER, FADEOUT_GRADIENT_HEIGHT)}
-        colors={[DEVICE_TEXT_COLOR, isApplied ? DEVICE_SCREEN_BACKGROUND_COLOR : DEVICE_TEXT_COLOR]}
-    />
+const deviceScreenStyle = prepareNativeStyle<{ isPaginationEnabled: boolean }>(
+    (utils, { isPaginationEnabled }) => ({
+        justifyContent: 'center',
+        alignItems: 'center',
+        minHeight: DEVICE_SCREEN_HEIGHT,
+        paddingVertical: isPaginationEnabled ? utils.spacings.large : 40,
+        maxWidth: DEVICE_SCREEN_WIDTH,
+        backgroundColor: DEVICE_SCREEN_BACKGROUND_COLOR,
+        borderRadius: utils.borders.radii.large / 2,
+    }),
 );
 
 export const UnverifiedAddressDevice = ({
     address,
     isAddressRevealed,
+    isCardanoAddress,
 }: DeviceScreenAddressProps) => {
     const { applyStyle } = useNativeStyles();
+    const [activePage, setActivePage] = useState<1 | 2>(1);
 
-    const addressChunkedLines = pipe(address, splitAddressToChunks, A.splitEvery(4));
+    const deviceModel = useSelector(selectDeviceModel);
 
-    const deviceFont = useFont(
-        require('../../../../packages/theme/fonts/RobotoMono-Regular.ttf'),
-        FONT_SIZE,
-    );
+    if (!deviceModel) return null;
+
+    const isPaginationEnabled = isCardanoAddress && isPaginationCompatibleDeviceModel(deviceModel);
+
+    const handlePaginationButtonPress = () => {
+        if (!isPaginationEnabled || !isAddressRevealed) return;
+        setActivePage(activePage === 1 ? 2 : 1);
+    };
+
+    const handlePaginationPanGesture = (panEvent: GestureEvent<PanGestureHandlerEventPayload>) => {
```

Comment:
You should definitely use useCallback for callbacks that you are passing to 3rd party libs because you don't know if lib doesn't rely on function stability

Reply:
fixed 45dd281869